addons/hr_timesheet/views/project_portal_templates.xml

```diff
@@ -41,8 +41,10 @@
 
     <template id="portal_tasks_list_inherit" inherit_id="project.portal_tasks_list" name="Portal: My Tasks with Timesheets">
         <xpath expr="//t[@t-call='portal.portal_table']" position="inside">
-            <t t-set="timesheet_ids" t-value="task.sudo().timesheet_ids"/>
-            <t t-set="is_uom_day" t-value="timesheet_ids._is_timesheet_encode_uom_day()"/>
+            <t t-if="task">
```

Comment:
After investigating, it seems the xpath is wrong, we have to add this code in https://github.com/odoo/odoo/blob/85a485034b90db866da720a8edc5aee83365be72/addons/project/views/project_portal_templates.xml#L153

So the fix should be something like this:
```diff
diff --git a/addons/hr_timesheet/views/project_portal_templates.xml b/addons/hr_timesheet/views/project_portal_templates.xml
index 88cac3f79c8..9fca59ff568 100644
--- a/addons/hr_timesheet/views/project_portal_templates.xml
+++ b/addons/hr_timesheet/views/project_portal_templates.xml
@@ -40,7 +40,7 @@
     </template>
 
     <template id="portal_tasks_list_inherit" inherit_id="project.portal_tasks_list" name="Portal: My Tasks with Timesheets">
-        <xpath expr="//t[@t-call='portal.portal_table']" position="inside">
+        <xpath expr="//t[@t-foreach='tasks']/tr" position="before">
             <t t-set="timesheet_ids" t-value="task.sudo().timesheet_ids"/>
             <t t-set="is_uom_day" t-value="timesheet_ids._is_timesheet_encode_uom_day()"/>
         </xpath>
```
in that case, we can be sure task is defined even if task contains no record.

What do you think? @IT-Ideas 🙂

Reply:
Can you also reword your commit message like this:
```txt
[FIX] hr_timesheet: fix traceback on task search in portal view

Before this commit, when the portal user is in `/my/tasks` route 
and he remove the groupby and search a certain task name and no 
task is found with his search, a traceback is occurred saying `task`
is not found to be able to evaluate the following lines:
```xml
                <t t-set="timesheet_ids" t-value="task.sudo().timesheet_ids"/>
                <t t-set="is_uom_day" t-value="timesheet_ids._is_timesheet_encode_uom_day()"/>
```
in the portal template.

This commit fixes the issue by changing the `xpath` to be sure the task
variable is defined before evaluating `task.sudo().timesheet_ids`.
Moreover, this commit checks if there is any task in `tasks` before 
displaying `tbody` to avoid having any scrollbar for nothing.

task-2987387
```
?
Thanks in advance! 🙂

Reply:
@xavierbol . 👍🏼 , done the changes. Thanks